text_handler.py

```diff
@@ -26,42 +29,58 @@ async def __handleYoutubeDownload(self, message: types.Message):
         if not str(message.chat.id) in config.WHITELIST_CHAT_ID:
             return
         try:
-            yt = YouTube(message.text)
-            author = yt.author
-            length = yt.length
-            title = yt.title
-            stream = (yt.streams
-                      .filter(only_audio=True, file_extension='mp4')
-                      .order_by('bitrate').desc().first())
-            filesize = stream.filesize
-            video_id = extract.video_id(message.text)
-            video_url = "youtu.be/" + video_id
+            ydl_opts = {
+                    'writethumbnail': True,
+                    'outtmpl': 'downloads/%(extractor)s_%(id)s.%(ext)s',
+                    'format': 'bestaudio',
+            }
+            with YoutubeDL(ydl_opts) as ydl:
```

Comment:
isn't `with` pointless here? you already have `try` above?
you can just do
```
            ydl = YoutubeDL(ydl_opts)
            info = ydl.extract_info(message.text, download=False)
            audio_url = format(info['url'])
```

Reply:
right. I didn't know about `try` yet.